### PR TITLE
Luke

### PR DESCRIPTION
**Pull request by luwo1701**
The artist search bar now drops down the an artists top 10 songs. We know have to figure out which drop down menu button was pressed so we can access that particular artists id

### CODE REVIEW

File: app/src/main/java/party_queue/myapplication/SpotifyActivity.java [hunk 1]
```diff
@@ -128,31 +129,63 @@ public void onError(Throwable throwable) {
...
         final Button button = (Button) v;
-        ( (Button) v).setText(search); // sets the text inside the button to be the text from the text box
+        //( (Button) v).setText(search); // sets the text inside the button to be the text from the text box
+        String type ="";
+        switch(button.getId()){
+            case R.id.button:
+                type = "track";
+                break;
+            case R.id.button2:
+                type = "artist";
```
Comment (cosh9041) [inline app/src/main/java/party_queue/myapplication/SpotifyActivity.java:146]: So this supports search by artist and search by song? Nice!

File: app/src/main/java/party_queue/myapplication/SpotifyActivity.java [hunk 2]
```diff
@@ -128,31 +129,63 @@ public void onError(Throwable throwable) {
...
+                                type = "track";
+                                try {
+                                    for (int i=0; i<10; i++) {
+                                        //artist = response.getJSONArray("artists").getJSONObject(0).getString("name");
+                                        songId[i] = response.getJSONObject(type+"s").getJSONArray("items").getJSONObject(i).getString("id");
+                                        artistnames[i] = response.getJSONObject(type+"s").getJSONArray("items").getJSONObject(i).getJSONArray("artists").getJSONObject(0).getString("name");
+                                        songName[i] = response.getJSONObject(type+"s").getJSONArray("items").getJSONObject(i).getString("name");
+                                    }
+
+                                } catch (JSONException e) {}
```
Comment (cosh9041) [inline app/src/main/java/party_queue/myapplication/SpotifyActivity.java:173]: We should probably at least print a message here in the catch block alerting if this exception has occured.

If this project moves forward past this semester, adding structured error logging/exception handling will be something we will want to do.

File: app/src/main/java/party_queue/myapplication/SpotifyActivity.java [hunk 3]
```diff
@@ -162,17 +195,81 @@ public void onResponse(JSONObject response) {
...
-                                                SpotifyActivity.this,
-                                                item.getTitle() + " added to queue",
-                                                Toast.LENGTH_SHORT
-                                        ).show();
+                                        final String title;
+                                        if (button.getId() == R.id.button) {
+                                            title = songName[0];
+                                            Toast.makeText(
+                                                    SpotifyActivity.this,
+                                                    "fuckit" + " added to queue",
```
Comment (cosh9041) [inline app/src/main/java/party_queue/myapplication/SpotifyActivity.java]: We might not want to merge this lol.

File: app/src/main/java/party_queue/myapplication/SpotifyActivity.java [hunk 4]
```diff
@@ -162,17 +195,81 @@ public void onResponse(JSONObject response) {
...
+                                                    Toast.LENGTH_SHORT
+                                            ).show();
+                                        }
+                                        //if song, add to q and set title = songname
+                                        else {
+
+                                            String id =  "43ZHCT0cAZBISjO8DG9PnE";
```
Comment (cosh9041) [inline app/src/main/java/party_queue/myapplication/SpotifyActivity.java]: Is this a user id or a spotify id?

**Comment by cosh9041** [comment]
This looks pretty good to me, nice work Luke. 

@busunkim96 @daniel-j-henderson Have either of y'all had a chance to look at this/import it into android studio and test it? We should try to get this merged in soon. My android studio is still broken, and I'm not sure it's worth investing the time to get it running at this point.

**Comment by daniel-j-henderson** [comment]
Looks good I can test tomorrow but it can merge fine

**Comment by cosh9041** [comment]
Let's hold off on merging until we get someone else to pull it down and get it running. I'm afraid environment issues due to android studio weirdness, and I'd like to get any of that ironed out before merging to master

**Pull request merged**